kapitan/targets.py

```diff
@@ -78,12 +79,23 @@ def compile_targets(
             search_paths=search_paths,
             compile_path=temp_path,
             ref_controller=ref_controller,
+            inventory_path=inventory_path,
             **kwargs,
         )
 
         if not target_objs:
             raise CompileError("Error: no targets found")
 
+        if kwargs.get("fetch_inventories", False):
+            # new_source checks for new sources in fetched inventory items
+            new_sources = list(set(list_sources(target_objs)) - cached.inv_sources)
+            while new_sources:
```

Comment:
I think it's cleaner to turn this into a for loop

Reply:
Since it is not known how many times the loop should run (as fetched inventories can fetch other inventories) I decide to use a while loop. I might be missing something but shouldn't a for loop be better only if it looped for a fixed number of times?